Packs/DuoAdminApi/Integrations/DuoEventCollector/DuoEventCollector.py

```diff
@@ -75,23 +81,59 @@ def call(self, request_order: list) -> tuple:
 
         return ([], [])
 
+    def check_window_before_call(self, mintime: int | float) -> bool:
+        """ Check if the API call should be performed. If not return false, else return true.
+            If the fetch_delay != 0 (we want a delayed fetch) and end_window < mintime -> we don't want to perform the fetch
+            at the moment.
+
+        Args:
+            mintime (int | float): The wanted fetch time in a timestamp.
+
+        Returns:
+            bool: False - don't perform the API call, True - perform the API call
+        """
+        demisto.debug(f'check_window_before_call {mintime=}')
+        mintime_dt = datetime.fromtimestamp(mintime)
+        if self.params.fetch_delay != '0' and self.params.end_window <= mintime_dt:
+            demisto.debug(f"check_window_before_call, don't perform API call {self.params.fetch_delay=} and "
+                          f"{self.params.end_window=} <= {mintime_dt=}")
+            return False
+        demisto.debug('check_window_before_call, perform API call')
+        return True
+
     def handle_authentication_logs(self) -> tuple:
         """
         Uses the V2 version of the API.
         For the first time the logs are retreived will work with mintime parameter.
         All other calls will be made with the next_offset parameter returned from the last fetch.
         get_authentication_log: If not provided takes mintime to 24 hours and maxtime to time.now - 2 min.
+
+        In case there is a fetch_delay, we want to perform the API call only in case that min_time is before the end of the
+        fetch window.
         """
+        maxtime = str(int(self.params.end_window.timestamp() * 1000))
+
         if not self.params.mintime[LogType.AUTHENTICATION].get('next_offset'):
-            response = self.admin_api.get_authentication_log(
-                mintime=self.params.mintime[LogType.AUTHENTICATION].get('min_time'),
-                api_version=2, limit=str(min(int(self.params.limit), int('1000'))), sort='ts:asc')
+            demisto.debug(f'handle_authentication_logs, no next_offset {maxtime=}')
+            try:
+                response = self.admin_api.get_authentication_log(
+                    mintime=self.params.mintime[LogType.AUTHENTICATION].get('min_time'),
+                    api_version=2, limit=str(min(int(self.params.limit), int('1000'))), sort='ts:asc', maxtime=maxtime)
+            except Exception as e:
```

Comment:
We need to ignore the errors in the case that `maxtime < mintime`, in case we are not in the fetch window. I will use a more specific exception statement.